src/fns/unconstrained_helpers.nr

```diff
@@ -230,32 +230,33 @@ pub(crate) unconstrained fn __barrett_reduction<let N: u32>(
     let mut quotient_u60 = mulout_u60.shr((k + k + BARRETT_REDUCTION_OVERFLOW_BITS));
 
     // N.B. we assume that the shifted quotient cannot exceed 2 times original bit size.
-    //      (partial_quotient should be just slightly larger than the modulus, we could probably represent with a size N+1 array)
+    //      (partial_quotient_full should be just slightly larger than the modulus, we could probably represent with a size N+1 array)
     let partial_quotient_full: [Field; 3 * N] = quotient_u60.into_field_array();
-    let mut partial_quotient: [Field; 2 * N] = [0; 2 * N];
-    for i in 0..2 * N {
```

Comment:
wait why was there a loop here? it seems it's just copying the array element by element

Reply:
Yeah this is what I first noticed. It looks to be a totally unnecessary reassignment.